corehq/apps/users/views/__init__.py

```diff
@@ -1198,7 +1199,12 @@ def role_to_choice(role):
 
     roles = UserRole.by_domain(domain)
     if not couch_user.is_domain_admin(domain):
-        roles = [role for role in roles if role.is_non_admin_editable]
+        user_role = couch_user.get_role()
```

Comment:
Can you wrap this in a try/except that sets `user_role` to None if a `DomainMembershipError` is encountered? Because of [this new feature](https://github.com/dimagi/commcare-hq/pull/27510) there are users who have access to domains but aren't admins and also don't have a role. That leads to https://sentry.io/organizations/dimagi/issues/1707410662/